### PR TITLE
Basic policy struct and parse

### PR DESCRIPTION
**Pull request by asraa**
Signed-off-by: Asra Ali <asraa@google.com>

#### Ticket Link

Fixes
* https://github.com/sigstore/sget/issues/15
#### Release Note

```release-note

```

### CODE REVIEW

File: src/policy.rs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    pub keyid: String,
+    // The base64 encoded signature of the canonical JSON of the root policy.
+    pub sig: String,
+    // The base64 encoded certificate that was used to create the signature.
+    pub cert: String,
+}
+
+// The root policy indicated the trusted root keys.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(tag = "_type")]
```
Comment (lkatalin) [inline src/policy.rs]: What does this tag do?

**Review by lkatalin — APPROVED** [review]
This looks good to me. :+1: Should we un-draft and merge it?

**Comment by lukehinds** [comment]
lgtm 2, are you ok to un-draft @asraa

**Comment by asraa** [comment]
I'm working through rust errors, It might be helpful if I push what I have but it seems like I've been having to implement deserialize/serialize methods for a lot of the types before my test can pass

**Comment by asraa** [comment]
Quick rust question -- the problem is that it doesn't detect `FromStr` for a `HashMap`. I've tried to use the `serde_with` to parse it from a json string and do the right thing, but I'm still having problems -- is there a way to do this without having to manually implement FromStr, it seems to me like it should understand from JSON and the structs that are in the hash map

```

error[E0277]: the trait bound `HashMap<RoleType, RoleKeys>: FromStr` is not satisfied
   --> src/policy.rs:40:5
    |
40  |     #[serde_as(as = "HashMap<serde_with::json::JsonString, serde_with::json::JsonString>")]
    |     ^ the trait `FromStr` is not implemented for `HashMap<RoleType, RoleKeys>`
    |
note: required by `std::str::FromStr::from_str`

error[E0277]: the trait bound `HashMap<std::string::String, Key>: FromStr` is not satisfied
   --> src/policy.rs:42:5
    |
42  |     #[serde_as(as = "HashMap<_, serde_with::json::JsonString>")]
    |     ^ the trait `FromStr` is not implemented for `HashMap<std::string::String, Key>`
    |
note: required by `std::str::FromStr::from_str`


```

**Comment by lkatalin** [comment]
@asraa Let me think about this and I'll maybe look at your branch. `serde` can be pretty tricky in Rust so you have jumped into the hard stuff first!

**Comment by asraa** [comment]
Done! I commented out the HashMap problem code above, and completed the test to just make sure version was being properly parsed. Hopefully this unblocks.

**Review by jyotsna-penumaka — APPROVED** [review]

**Pull request merged**